client/fingerprint/consul.go

```diff
@@ -88,46 +109,61 @@ func (f *ConsulFingerprint) Periodic() (bool, time.Duration) {
-		}
+func (cfs *consulFingerprintState) initialize(cfg *config.ConsulConfig, logger hclog.Logger) error {
+	if cfs.client != nil {
+		return nil // already initialized!
+	}
 
-		f.client, err = consulapi.NewClient(consulConfig)
-		if err != nil {
-			return fmt.Errorf("failed to initialize Consul client: %s", err)
-		}
+	consulConfig, err := cfg.ApiConfig()
+	if err != nil {
+		return fmt.Errorf("failed to initialize Consul client config: %v", err)
+	}
+
+	cfs.client, err = consulapi.NewClient(consulConfig)
+	if err != nil {
+		return fmt.Errorf("failed to initialize Consul client: %s", err)
+	}
 
-		f.extractors = map[string]consulExtractor{
-			"consul.server":        f.server,
-			"consul.version":       f.version,
-			"consul.sku":           f.sku,
-			"consul.revision":      f.revision,
-			"unique.consul.name":   f.name,
-			"consul.datacenter":    f.dc,
-			"consul.segment":       f.segment,
-			"consul.connect":       f.connect,
-			"consul.grpc":          f.grpc(consulConfig.Scheme),
-			"consul.ft.namespaces": f.namespaces,
+	if cfg.Name == "default" {
+		cfs.extractors = map[string]consulExtractor{
+			"consul.server":        cfs.server,
+			"consul.version":       cfs.version,
+			"consul.sku":           cfs.sku,
+			"consul.revision":      cfs.revision,
+			"unique.consul.name":   cfs.name, // note: won't have this for non-default clusters
+			"consul.datacenter":    cfs.dc,
+			"consul.segment":       cfs.segment,
+			"consul.connect":       cfs.connect,
+			"consul.grpc":          cfs.grpc(consulConfig.Scheme, logger),
+			"consul.ft.namespaces": cfs.namespaces,
+		}
+	} else {
+		cfs.extractors = map[string]consulExtractor{
+			fmt.Sprintf("consul.%s.server", cfg.Name):        cfs.server,
+			fmt.Sprintf("consul.%s.version", cfg.Name):       cfs.version,
+			fmt.Sprintf("consul.%s.sku", cfg.Name):           cfs.sku,
+			fmt.Sprintf("consul.%s.revision", cfg.Name):      cfs.revision,
+			fmt.Sprintf("consul.%s.datacenter", cfg.Name):    cfs.dc,
+			fmt.Sprintf("consul.%s.segment", cfg.Name):       cfs.segment,
+			fmt.Sprintf("consul.%s.connect", cfg.Name):       cfs.connect,
+			fmt.Sprintf("consul.%s.grpc", cfg.Name):          cfs.grpc(consulConfig.Scheme, logger),
+			fmt.Sprintf("consul.%s.ft.namespaces", cfg.Name): cfs.namespaces,
 		}
 	}
 
 	return nil
 }
 
-func (f *ConsulFingerprint) query(resp *FingerprintResponse) agentconsul.Self {
+func (cfs *consulFingerprintState) query(resp *FingerprintResponse, logger hclog.Logger) agentconsul.Self {
 	// We'll try to detect consul by making a query to to the agent's self API.
 	// If we can't hit this URL consul is probably not running on this machine.
-	info, err := f.client.Agent().Self()
+	info, err := cfs.client.Agent().Self()
 	if err != nil {
 		// indicate consul no longer available
-		if f.lastState == consulAvailable {
-			f.logger.Info("consul agent is unavailable")
+		if cfs.isAvailable {
+			logger.Info("consul agent is unavailable")
```

Comment:
Is it worth adding the error to this log line?

Reply:
Probably can't hurt. Will do.